.clang-format

```diff
@@ -0,0 +1,31 @@
+---
```

Comment:
I would prefer to move it into /src/
Does the github action support this?

Reply:
The github action can support it, but this would require configuration in editors. I'm not sure how to do that in msvs (and eventually with cmake in #7722)

Reply:
MSVS detects it automatically (at least during my tests).
How is cmake related to editors?

Reply:
If msvs required configuration, then the next step would have been to add it to the solution. and then when the cmake changes go in, how to do it there too. But if msvs doesn't need any special config, then we should be fine